src/shared-components/progressBar/index.tsx

```diff
@@ -4,18 +4,34 @@ import PropTypes from 'prop-types';
 import { COLORS, PROGRESS_BAR_STATUS } from '../../constants';
 import { OuterContainer, InnerBar } from './style';
 
+export type ProgressBarStatusType =
```

Comment:
I think this might be easier if we pull over the `valueof` utility into this repo.

```ts
type valueof<T> = T[keyof T];
type ProgressBarStatusTypes = valueof<typeof PROGRESS_BAR_STATUS>;
```

Reply:
it worked with `export type ProgressBarStatusType = keyof (typeof PROGRESS_BAR_STATUS);`
thanks I was looking for an example to reduce the lines

Reply:
@daigof That works because the keys happen to match the values, but we should be sure to target the literal values instead of the keys. So you can just in-line the `valueof` definition: `typeof PROGRESS_BAR_STATUS[keyof typeof PROGRESS_BAR_STATUS]` should work.

Reply:
Yeah I understand about key thing. 
I was honestly looking for a better way to define my type, Ill push the version you provided but it looks kind of weird sort of for readability purpose or short list of props the first way I coded look better maybe?

I want to have this at hand for future uses so I want to commit your version

Reply:
I agree it's not very readable, so we should probably add the `valueof` to be able to be used everywhere like we do in our main repo